src/Cns.ts

```diff
@@ -57,14 +54,37 @@ export default class Cns extends NamingService {
     );
   }
 
+  static async getNetworkConfigs(config?: { url: string } | { provider: Provider } ): Promise<{ network: CnsSupportedNetworks, provider: Provider } | undefined> {
+    if (!config) {
+      return undefined;
+    }
+    try {
+      const provider = hasProvider(config) ? config.provider : new FetchProvider(NamingServiceName.CNS, config.url);
```

Comment:
Please check `config.url` existence/emptiness before you're setting up new FetchProvider()